QUARKUS-6292.md

```diff
@@ -0,0 +1,90 @@
+# QUARKUS-6292 - Extend the platform with offering
+
+JIRA: https://issues.redhat.com/browse/QUARKUS-6292
+
+Extend the platform with offering
+
+Upstream PRs:
+- https://github.com/quarkusio/quarkus/pull/49589
+- https://github.com/quarkusio/quarkus/pull/49487
+- https://github.com/redhat-developer/code.quarkus.redhat.com/pull/190
+- https://github.com/redhat-developer/code.quarkus.redhat.com/issues/183
+- https://github.com/quarkusio/code.quarkus.io/pull/1656
+- https://github.com/quarkusio/code.quarkus.io/pull/1652
+
+Documentation:
+- https://quarkus.io/guides/extension-metadata#extension-offerings
+- https://quarkus.io/guides/extension-registry-user#limiting-extension-catalog-to-an-offering
+
+## Scope of testing
+QUARKUS-6292 start dividing the offering by adding the offering metadata each supported extension.
+The metadata label should have format of `<offering-id>-support`, where the extension can have 0-n labels.
+
+As adding the multiple offering labels, there are 3 category which should be adjusted/checked:
+1. Quarkus registry (registry.quarkus)
+2. Code Quarkus (code.quarkus)
+3. Quarkus CLI
+
+### Quarkus registry (registry.quarkus)
+
+For registry there is no need for any update as the offering labels are part of already existing metadata.
+Currently only offering label in metadata is `redhat-support`.
+
+### Code Quarkus (code.quarkus)
+
+Update of Code Quarkus is defined in https://issues.redhat.com/browse/QUARKUS-6293.
+There will be separate Code Quarkus for each offering (e.g. each offering will have own URL).
+
+In addition, the Code Quarkus add possibility to search for extension with some scope of support or search only for the extension without any support.
+
+### Quarkus CLI
+
+The Quarkus CLI is defined in https://issues.redhat.com/browse/QUARKUS-6397. 
+The Quarkus CLI should be able to show which extension is part of which offering, when listing the extension.
+Documentation about definition and usage can be found in references section of this test plan.
+
+At the moment of Quarkus 3.27 the only way to add the offering is to add it to `.quarkus/config.yaml` manually.
+There is plan to add the offering using the Quarkus CLI command (https://github.com/quarkusio/quarkus/issues/49090)
+
+
+## Automated test development
+
+### Code Quarkus (code.quarkus)
+
+The start-stop testsuite already contains the tests for Code Quarkus.
+These tests are cover only RedHat Code Quarkus, which will be used to test `redhat` offering.
+The tests for IBM Code Quarkus needs to be added.
+The IBM tests will be modification of the `CodeQuarkusSiteTest`.
+Also, the generic tests for Code Quarkus in `CodeQuarkusTest` will be executed for both RedHat and IBM Code Quarkus.
```

Comment:
Can you write a bit more about how the `CodeQuarkusTest` will be modified to test both?

Will this need job updates too? If yes, maybe this is worth mentioning also.

Reply:
> Can you write a bit more about how the `CodeQuarkusTest` will be modified to test both?
In general the `CodeQuarkusTest` wont be updated as its testing download of the project and executing it. It downloading it from url which can be passed by property. I added this to TP

> Will this need job updates too? If yes, maybe this is worth mentioning also.
Added to the TP